client/me/memberships/subscription.jsx

```diff
@@ -33,42 +33,67 @@ function Subscription( { translate, subscription, moment, stoppingStatus } ) {
 	useEffect( () => {
 		if ( stoppingStatus === 'fail' ) {
 			// run is-error notice to contact support
-			dispatch(
-				errorNotice(
-					translate(
-						'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
-						{
-							components: {
-								a: <a href={ CALYPSO_CONTACT } />,
-								strong: <strong />,
-							},
-						}
-					)
-				)
-			);
+			subscription && ! subscription.renew_interval
+				? dispatch(
+						errorNotice(
+							translate(
+								'There was a problem while removing your product, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
+								{
+									components: {
+										a: <a href={ CALYPSO_CONTACT } />,
+										strong: <strong />,
+									},
+								}
+							)
+						)
+				  )
+				: dispatch(
+						errorNotice(
+							translate(
+								'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
+								{
+									components: {
+										a: <a href={ CALYPSO_CONTACT } />,
+										strong: <strong />,
+									},
+								}
+							)
+						)
+				  );
 		} else if ( stoppingStatus === 'success' ) {
 			// redirect back to Purchases list
 			dispatch(
-				successNotice(
-					translate( 'This subscription has been canceled. You will no longer be charged.' ),
-					{ displayOnNextPage: true }
-				)
+				successNotice( translate( 'This item has been removed.' ), { displayOnNextPage: true } )
```

Comment:
Is there a reason why we are displaying this for every case?
Couldn't we do, for example:

```js
subscription && ! subscription.renew_interval
  ? successNotice( translate( 'This item has been removed.' ), { displayOnNextPage: true } )
  : successNotice( translate( 'This subscription has been canceled. You will no longer be charged.' ), { displayOnNextPage: true } )
```

I mean I think that "you will no longer be charged" is valuable information for subscription products.

Reply:
Also, we should probably be consistent and use "product" instead of "item".

```suggestion
				successNotice( translate( 'This product has been removed.' ), { displayOnNextPage: true } )
```

Reply:
> Is there a reason why we are displaying this for every case? Couldn't we do, for example:
> 
> ```js
> subscription && ! subscription.renew_interval
>   ? successNotice( translate( 'This item has been removed.' ), { displayOnNextPage: true } )
>   : successNotice( translate( 'This subscription has been canceled. You will no longer be charged.' ), { displayOnNextPage: true } )
> ```
> 
> I mean I think that "you will no longer be charged" is valuable information for subscription products.

I think the subscription is no longer available in the state because it's been removed, so we cannot actually know whether it was a one-time plan or not. We discussed this briefly yesterday with @BogdanUngureanu and we agreed on simplifying the message to something generic that works for all types of subscriptions, rather than refactoring how actions are dispatched in order to propagate the subscription info.